client/ayon_core/hosts/resolve/api/lib.py

```diff
@@ -899,13 +928,53 @@ def get_otio_clip_instance_data(otio_timeline, timeline_item_data):
 
             # add pypedata marker to otio_clip metadata
             for marker in otio_clip.markers:
-                if self.pype_marker_name in marker.name:
+                if constants.ayon_marker_name in marker.name:
                     otio_clip.metadata.update(marker.metadata)
             return {"otioClip": otio_clip}
 
     return None
 
 
+def get_timeline_otio_filepath(project_name, anatomy=None, timeline=None):
+    """Get timeline otio filepath.
+
+    Args:
+        project_name (str): ayon project name
+        anatomy (ayon_core.pipeline.Anatomy)[optional]: Anatomy object
+        timeline (resolve.Timeline)[optional]: resolve's object
+
+    Returns:
+        str: temporary otio filepath
+    """
+    from . import bmdvr
+    resolve_project = get_current_resolve_project()
+    timeline = resolve_project.GetCurrentTimeline()
+    timeline_name = timeline.GetName()
```

Comment:
This should be using the `timeline` that's passed along as an argument if not `None`.
```suggestion
    if timeline is None:
        timeline = resolve_project.GetCurrentTimeline()
        if not timeline:
            raise RuntimeError("No current timeline")
    timeline_name = timeline.GetName()
```